TypeAnot.py

```diff
@@ -0,0 +1,89 @@
+from typing import Any, List, Dict, Union, Optional
+
+
+def format_name(first_name: str, last_name: str) -> str:
+    """Return a full name: "John JAMESEON"."""
+    a = first_name.capitalize()
+    b = last_name.upper()
+    return f"{a} {b}"
+
+
+def format_name_annotated(first_name: str, last_name: str) -> str:
+    """Return a full name: "John JAMESEON"."""
+    a = first_name.capitalize()
+    b = last_name.upper()
+    return f"{a} {b}"
+
+
+def process_character(s: str) -> None:
+    pass
+
+
+def add_words_to_collection(string_to_process: str, collection: List[str]) -> str:
+    for word in string_to_process.split():
+        collection.append(word)
+    return collection[0]
+
+
+def add_words_to_collection_annotated(string_to_process: str, collection: List[str]) -> str:
+    for word in string_to_process.split():
+        collection.append(word)
+    return collection[0]
+
+
+def concatenate(s1: str, s2: str) -> str:
+    return s1 + s2
+
+
+def is_even(x: int) -> bool:
+    return x % 2 == 0
+
+
+def greet(name: str, greeting: Optional[str] = "Hello") -> str:
+    return f"{greeting}, {name}!"
+
+
+def parse_number(num: Union[int, float, str]) -> Union[int, float, None]:
```

Comment:
I can't agree with you. converting float to int, isn't parsing. 
Parse means that we parse string and have a numeric type in the output (or None).
Type of num should be just str.